OpenEdXMobile/src/main/java/org/edx/mobile/deeplink/PushLinkManager.kt

```diff
@@ -3,25 +3,40 @@ package org.edx.mobile.deeplink
 import android.content.Context
 import android.content.Intent
 import android.os.Bundle
+import com.google.firebase.messaging.RemoteMessage
+import de.greenrobot.event.EventBus
+import org.edx.mobile.event.PushLinkReceivedEvent
+import org.edx.mobile.logger.Logger
 
 object PushLinkManager {
+    val logger = Logger(this.javaClass)
+
     fun checkIfReceiveNotification(context: Context, intent: Intent?) {
         val screenName = intent?.extras?.getString(DeepLink.Keys.SCREEN_NAME)
         if (screenName != null && screenName.isNotEmpty()) {
-            // Check if launcher activity started from FCM notification (only possible when app will be in background state)
-            didReceiveFCMNotification(context, screenName, intent.extras, false)
+            // Check if launcher activity started from FCM notification (only possible when app will be in background state as per doc)
+            didReceiveFCMBackgroundNotification(context, screenName, intent.extras)
         }
     }
 
-    private fun didReceiveFCMNotification(context: Context, screenName: String, bundle: Bundle, isForeground: Boolean) {
-        didReceiveRemoteNotification(context, DeepLink(screenName, bundle), isForeground)
+    private fun didReceiveFCMBackgroundNotification(context: Context, screenName: String, bundle: Bundle) {
```

Comment:
We can use the common method by refactoring any method **(userGrantedPushLinkAction/ didReceiveFCMBackgroundNotification)**.

Reply:
As per discussion, existing code is more scaleable